torch/csrc/jit/graph_executor.cpp

```diff
@@ -114,11 +125,10 @@ struct ExecutionPlan {
 
 private:
   // inplace to avoid allocations
-  variable_tensor_list unwrapVariables(variable_tensor_list && list) const {
-    for(auto & v : list) {
-      v = v.defined() ? autograd::as_variable_ref(v).detach() : at::Tensor();
-    }
-    return std::move(list);
+  std::vector<IValue> unwrapVariables(variable_tensor_list && list) const {
+    return fmap(list, [](const Variable& v) -> IValue {
+      return v.defined() ? autograd::as_variable_ref(v).detach() : at::Tensor();
+    });
```

Comment:
1. The comment above is no longer true.
2. Why do we take an r-value reference here? a const ref would work perfectly fine if we're reallocating anyway

Reply:
Updated the comment. I want to keep the r-value here, because I hope to eventually figure out a way to keep the types lined up enough to avoid the copies that this patch introduces. Thoughts:

1. Use ShortVector so most of time we don't get allocations. Downsize: the copies/moves are expensive
2. Switch Autograd to store IValue rather than Tensor. Downside: potential overhead in autograd (though maybe avoidable if we have autograd use unchecked versions of toTensor internally).
3. Use our own vector type that can safely move its data array from compatible types. Prereq: Tensor/IValue need to have the same in-memory representation, or at least the same size.

For now I am not going to do any of these since they will take time and it is not clear it is a critical pathway. We should follow up with @zou3519 when he figures out what the slowness is for LSTMs but so far it seems unrelated to the executor and exists inside of aten.